core/testing/include/Kiso_Testing.h

```diff
@@ -0,0 +1,192 @@
+ *                  Kiso_Testing library.
+*/
+enum KISO_TESTING_Retcodes_E
+{
+    RETCODE_TESTING_SUITE_ALREADY_REGISTERED = RETCODE_FIRST_CUSTOM_CODE,
+    RETCODE_TESTING_CASE_ALREADY_REGISTERED,
+    RETCODE_TESTING_CCHANNEL_INITIALISATION_FAILED,
+    RETCODE_TESTING_INCOMPLETE_MESSAGE_RECEIVED,
+    RETCODE_TESTING_CRC_MISMATCH,
+    RETCODE_TESTING_VERSION_MISMATCH,
+    RETCODE_TESTING_TLVELEMENT_NOT_FOUND,
+    RETCODE_TESTING_REPORT_TIMEOUT,
+    RETCODE_TESTING_CCHANNEL_NOT_SPECIFIED,
+};
+
+/**
+ * @brief           Enumerates the types of messages exchanged between the Test_Executor and the Test_Controller 
+ */
+enum CCMsg_MessageType_E
+{
+    CCMSG_TYPE_COMMAND,
+    CCMSG_TYPE_REPORT,
+    CCMSG_TYPE_ACK,
+};
+
+/**
+ * @brief           Encapsulates the elements composing the message header.
+ */
+typedef struct MessageHeader_S
+{
+    uint8_t messageInfo; /**< version is 2 bits, message type is 2 bits and the remaining 4 bits are reserved */
+    uint8_t messageToken;
+    uint8_t messageType;
+    uint8_t errorCode;
+    uint8_t testEntry;
+    uint8_t testSuite;
+    uint8_t testCase;
+    uint8_t payloadLength;
+} MsgHdr_T;
+
+/**
+ * @brief           Encapsulates the elements composing a TLV object(type, length and value)
+ */
+typedef struct TlvElement_S
+{
+    uint8_t type;
+    uint8_t length;
+    char *value;
+} TlvElt_T;
+
+/**
+ *  @brief          Encapsulates the elements composing a message..
+ */
+typedef struct CCMsg_S
+{
+    MsgHdr_T header; /**< Header of the message */
+    uint8_t payload[TEST_MAX_PAYLOAD_LENGTH];
+    TlvElt_T tlvArray[CCHANNEL_MAX_NUMBER_OF_TLV_ELEMENTS]; /**< Parsed TLV array (from receive buffer) */
+    uint8_t payloadIndex;
+    uint8_t tlvArrayIndex;
+    uint8_t numberOfTlvElements; /**< Number of TLV elements for the message */
+    uint8_t rebootCounter;
+    bool isFree : 1; /**< This bit indicates whether the received message is processed by test runner or not */
+} CCMsg_T;
+
+/**
+ * @brief           Defines a prototype type for the setup functions of the test suites and test cases.
+ */
+typedef Retcode_T (*SetupFct_T)(CCMsg_T *ccmsg);
+
+/**
+ * @brief           Defines a prototype type for the tear down functions of the test suites and test cases.
+ */
+typedef Retcode_T (*TearDownFct_T)(CCMsg_T *ccmsg);
+
+/**
+ * @brief           Sefines a prototype type for the run functions of the test cases.
+ */
+typedef void (*RunFct_T)(CCMsg_T *ccmsg);
+
+/*###################### EXPORTED FUNCTIONS PROTOTYPES ###############################################################*/
+
+/**
+ * @brief           Initializes the Testing Framework.
+ * @details         this
+ * @param[in]       eId Id of the Test Entry
+ * @return          RETCODE_OK if initialized successfully error code otherwise
+ */
+Retcode_T Tests_Initialize(uint8_t eId, SetupFct_T setup, TearDownFct_T teardown);
```

Comment:
this bothers me: while writing code, i will have to lookup what is behind eId, even though make the implementation hard to read.